### ISSUE TITLE
[A11y] 1.1.1 Non-Text Content - Form Inputs Lack Text Labels

### ISSUE THREAD

**Issue by dalonsoa** [research-software-directory/RSD-as-a-service#1747]
See https://github.com/ImperialCollegeLondon/RSD-as-a-service/issues/59 for more details

**Comment by dalonsoa** [comment]
I'm taking this one, now.

**Comment by dalonsoa** [comment]
It seems this is a built-in issue with MUI, and hence of no easy solution at all :( . See https://stackoverflow.com/questions/77555283/mui-select-component-warnings-no-label-no-label-associated-with-a-form-field-h

It seems it also happens for the `RadioGroup` and while the issue was observed about two years ago, only now they have started to work on it. https://github.com/mui/material-ui/issues/40860

IT would seem that MUI itself is not designed to comply - or let you comply - with the accessibility standards.

@ewan-escience do you have any suggestion about this?

**Comment by ewan-escience** [comment]
I'm not sure what exactly the problem is. On [the Mozilla documentation site on labels](https://developer.mozilla.org/en-US/docs/Web/HTML/Reference/Elements/label), they have the following code snippet:

```html
<label>
  I like peas.
  <input type="checkbox" name="peas" />
</label>
```

So you can have an `input` inside of a `label`. No `id` or `for` needed.

On the [MUI docs](https://mui.com/material-ui/react-checkbox/), there is a seemingly equivalent code snippet:

```jsx
import FormGroup from '@mui/material/FormGroup';
import FormControlLabel from '@mui/material/FormControlLabel';
import Checkbox from '@mui/material/Checkbox';

export default function CheckboxLabels() {
  return (
    <FormGroup>
      <FormControlLabel control={<Checkbox defaultChecked />} label="Label" />
      <FormControlLabel required control={<Checkbox />} label="Required" />
      <FormControlLabel disabled control={<Checkbox />} label="Disabled" />
    </FormGroup>
  );
}
```

Does that help?

**Comment by dalonsoa** [comment]
I think that the problem is that such MUI form control does not translate into an HTML that looks like the Mozilla example you indicate.

For example, the following code in the `CiteDropdown.tsx` file:

https://github.com/research-software-directory/RSD-as-a-service/blob/2a2ac1cc6e6a766cf8c99173ebd5e8565202b6d6/frontend/components/software/CiteDropdown.tsx#L23-L52

results in:

<img width="579" height="416" alt="Image" src="https://github.com/user-attachments/assets/d1cfe035-4452-4a05-afdb-3c2f82f4dcda" />

The `input` is not within the `label` and nothing of I've tried to implement `for` and `id` that relate both works.

**Comment by ewan-escience** [comment]
I tried to fix it with using the `htmlFor` attribute on the `InputLabel` (and enabling the `id` attribute on the `Select` component), but then the label refers to a `div` instead if the `input`. @dmijatovic do you have any idea?

**Comment by ewan-escience** [comment]
I found out the problem: the passed `label` in `frontend/components/software/CiteDropdown.tsx` contains spaces. Under the hood, an `aria-labelledby` is used and according to the [Mozilla docs](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Reference/Attributes/aria-labelledby):

> The aria-labelledby property takes as value a space-separated id reference list, which means you can combine more than one element into a single accessible name.

Hence, the `id` of the `InputLabel` and the `labelId` of `Select` should not contain any white spaces.

**Comment by ewan-escience** [comment]
You could e.g. use ``id={`cite-dropdown-${label}`.replaceAll(' ', '_')}`` and ``labelId={`cite-dropdown-${label}`.replaceAll(' ', '_')}``

**Comment by dalonsoa** [comment]
Many thanks for digging the solution! I'll try this approach.

**Comment by dmijatovic** [comment]
@ewan-escience and @dalonsoa I was too busy lately to do more search on this item but I found some time now. 

I'll make PR with proposed fix. There seem to be specific approach which should be followed in order to get `aria-labelledby` correctly. I missed this approach too :-(.

### How to fix
The accessibility error occurs because dynamic id and labelId strings contain spaces if the label prop has spaces. To fix use:
- useId() hook: Automatically creates a unique, safe string ID per component instance. It prevents spaces from breaking WCAG attributes.
- Add id property: Select component requires a specific id along with its labelId so MUI can correctly map the hidden <input> tag to the floating element.
- inputProps fallback: Adding aria-label directly into the underlying input guarantees automated accessibility tools can calculate the field's name correctly.

**I assume that this approach should be applied on more locations where we use Select component.**
